src/System.DirectoryServices.Protocols/src/System/DirectoryServices/Protocols/common/ResultCode.cs

```diff
@@ -57,61 +55,59 @@ public enum ResultCode
+                { ResultCode.ReferralV2, SR.LDAP_PARTIAL_RESULTS },
+                { ResultCode.Referral, SR.LDAP_REFERRAL },
+                { ResultCode.AdminLimitExceeded, SR.LDAP_ADMIN_LIMIT_EXCEEDED },
+                { ResultCode.UnavailableCriticalExtension, SR.LDAP_UNAVAILABLE_CRIT_EXTENSION },
+                { ResultCode.ConfidentialityRequired, SR.LDAP_CONFIDENTIALITY_REQUIRED },
+                { ResultCode.SaslBindInProgress, SR.LDAP_SASL_BIND_IN_PROGRESS },
+                { ResultCode.NoSuchAttribute, SR.LDAP_NO_SUCH_ATTRIBUTE },
+                { ResultCode.UndefinedAttributeType, SR.LDAP_UNDEFINED_TYPE },
+                { ResultCode.InappropriateMatching, SR.LDAP_INAPPROPRIATE_MATCHING },
+                { ResultCode.ConstraintViolation, SR.LDAP_CONSTRAINT_VIOLATION },
+                { ResultCode.AttributeOrValueExists, SR.LDAP_ATTRIBUTE_OR_VALUE_EXISTS },
+                { ResultCode.InvalidAttributeSyntax, SR.LDAP_INVALID_SYNTAX },
+                { ResultCode.NoSuchObject, SR.LDAP_NO_SUCH_OBJECT },
+                { ResultCode.AliasProblem, SR.LDAP_ALIAS_PROBLEM },
+                { ResultCode.InvalidDNSyntax, SR.LDAP_INVALID_DN_SYNTAX },
+                { ResultCode.AliasDereferencingProblem, SR.LDAP_ALIAS_DEREF_PROBLEM },
+                { ResultCode.InappropriateAuthentication, SR.LDAP_INAPPROPRIATE_AUTH },
+                { ResultCode.InsufficientAccessRights, SR.LDAP_INSUFFICIENT_RIGHTS },
+                { ResultCode.Busy, SR.LDAP_BUSY },
+                { ResultCode.Unavailable, SR.LDAP_UNAVAILABLE },
+                { ResultCode.UnwillingToPerform, SR.LDAP_UNWILLING_TO_PERFORM },
+                { ResultCode.LoopDetect, SR.LDAP_LOOP_DETECT },
+                { ResultCode.SortControlMissing, SR.LDAP_SORT_CONTROL_MISSING },
+                { ResultCode.OffsetRangeError, SR.LDAP_OFFSET_RANGE_ERROR },
+                { ResultCode.NamingViolation, SR.LDAP_NAMING_VIOLATION },
+                { ResultCode.ObjectClassViolation, SR.LDAP_OBJECT_CLASS_VIOLATION },
+                { ResultCode.NotAllowedOnNonLeaf, SR.LDAP_NOT_ALLOWED_ON_NONLEAF },
+                { ResultCode.NotAllowedOnRdn, SR.LDAP_NOT_ALLOWED_ON_RDN },
+                { ResultCode.EntryAlreadyExists, SR.LDAP_ALREADY_EXISTS },
+                { ResultCode.ObjectClassModificationsProhibited, SR.LDAP_NO_OBJECT_CLASS_MODS },
+                { ResultCode.ResultsTooLarge, SR.LDAP_RESULTS_TOO_LARGE },
+                { ResultCode.AffectsMultipleDsas, SR.LDAP_AFFECTS_MULTIPLE_DSAS },
+                { ResultCode.VirtualListViewError, SR.LDAP_VIRTUAL_LIST_VIEW_ERROR },
+                { ResultCode.Other, SR.LDAP_OTHER }
+            };
         }
 
-        static public string MapResultCode(int errorCode)
-        {
-            return (string)s_resultCodeHash[(ResultCode)errorCode];
-        }
+        static public string MapResultCode(int errorCode) => (string)s_resultCodeHash[(ResultCode)errorCode];
```

Comment:
If errorCode isn't in the table, doesn't this now throw an exception whereas previously it returned null?